### PR TITLE
Team-only support: conversation access mode support

### PR DESCRIPTION
**Pull request by mikeger**
## Missing parts:

- [x] Support for the most recent BE status.
- [x] Migration test.
- [x] Payload loading.
- [x] Conversation creation API.

### CODE REVIEW

File: Tests/Source/Model/Conversation/ZMConversationTests+AccessMode.swift [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    var sut: ZMConversation!
+    
+    override func setUp() {
+        super.setUp()
+        sut = conversation()
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        sut = nil
```
Comment (vytis) [inline Tests/Source/Model/Conversation/ZMConversationTests+AccessMode.swift]: Tear down should be called at the end of the method

File: Tests/Source/Model/Conversation/ZMConversationTests+AccessMode.swift [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    }
+    
+    func testThatItCanSetTheMode() {
+        // when
+        sut.accessMode = .allowGuests
```
Comment (vytis) [inline Tests/Source/Model/Conversation/ZMConversationTests+AccessMode.swift]: I guess worth checking if access mode is different before this

File: Tests/Source/Model/Conversation/ZMConversationTests+AccessMode.swift [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            XCTAssertEqual(sut.allowGuests, $0.0)
+        }
+    }
+    
+    func testThatTheConversationIsInsertedWithCorrectAccessModeAccessRole_Default() {
+        // when
+        let conversation = ZMConversation.insertGroupConversation(into: self.uiMOC,
+                                                                  withParticipants: [],
+                                                                  name: "Test Conversation",
+                                                                  in: nil)!
```
Comment (vytis) [inline Tests/Source/Model/Conversation/ZMConversationTests+AccessMode.swift]: What should be the behaviour when we create conversations outside of the team? Maybe it should default to something different than inside teams?

Comment (mikeger) [reply Tests/Source/Model/Conversation/ZMConversationTests+AccessMode.swift]: Good point, I need to research is it safe to set the access mode of `code`, `invite` for the non-team user's conversations.

**Comment by mikeger** [comment]
@vytis the default mode is now defined by the backend: if user specifies the team when creating the conversation the access mode is set, otherwise it's nil and the backend is going to set it to the default values.

**Pull request merged**